src/planning/planner.ts

```diff
@@ -26,18 +26,14 @@ function _createTarget(
     targetType: interfaces.TargetType,
     serviceIdentifier: interfaces.ServiceIdentifier<any>,
     name: string,
-    key?: string | number | symbol,
```

Comment:
this will be a breaking change, we might want to think about this a bit

Reply:
Isn't this just a private method?

Reply:
It is private.  The plan method isn't but it is internal.  Somehow there isn't an existing test in planner.test.ts or resolver.test.ts that tests for this additional metadata.  Perhaps this pr would be an ideal opportunity.

Reply:
I'll add those.